test/sql-luatest/constraint_test.lua

```diff
@@ -407,3 +412,33 @@ g.test_constraints_5 = function()
         box.func.check_T_ONE:drop()
     end)
 end
+
+--
+-- Make sure that the SQL foreign key and check constraints are correctly
+-- converted to core foreign key and check constraints during the upgrade.
+--
+g.test_constraints_5 = function()
+    g.upgrade:exec(function()
+        local t = require('luatest')
+
+        t.assert_equals(box.space._ck_constraint:select(), {})
+        t.assert_equals(box.space._fk_constraint:select(), {})
+
+        local s = box.space._space.index[2]:get('T')
+        local s1 = box.space._space.index[2]:get('T1')
+        local s2 = box.space._space.index[2]:get('T2')
+        t.assert_equals(s[6].constraint, nil)
+        local fk1 = {fk_unnamed_T_1 = {field = {0}}}
+        t.assert_equals(s[6].foreign_key, fk1)
+        t.assert_equals(s1[6].constraint, nil)
+        local fk2 = {fk_unnamed_T1_1 = {field = {0}, space = s.id},
+            fk_unnamed_T1_2 = {field = {[0] = 1, [1] = 0}, space = s.id}}
+        t.assert_equals(s1[6].foreign_key, fk2)
+        local ck = {}
+        ck.ck_unnamed_T2_1 = box.func.check_T2_ck_unnamed_T2_1.id
+        ck.ck_unnamed_T2_2 = box.func.check_T2_ck_unnamed_T2_2.id
+        ck.ck_unnamed_T2_3 = box.func.check_T2_ck_unnamed_T2_3.id
+        t.assert_equals(s2[6].constraint, ck)
+        t.assert_equals(s2[6].foreign_key, nil)
+    end)
+end
```

Comment:
I suggest to add the script which you used to make the snapshot. Like `test/vinyl/upgrade/*/fill.lua`. Also are you use the folder should be 2.8.3? Last release was 2.10.0 if I remember correctly.

Reply:
Thanks, done!